app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -505,15 +517,34 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote, val w
 			}
 		} else {
 			loadPaginatedItems(directory.toListItem(), { !deferred.isCancelled }) {
-				deferred.complete(it)
+				val items = removeShufflePlayButtonMetadata(it)
+				deferred.complete(items)
 			}
 		}
 		return deferred.await()
 	}
 
-	override suspend fun search(query: String): List<MusicMetadata>? {
-		// requires a Web API call, not sure how to get the access token
-		return null
+	/**
+	 * Removes the shuffle play button [MusicMetadata] if it is present in the supplied list.
+	 *
+	 * When loading a list of tracks such as an album, a shuffle play button [MusicMetadata] object is
+	 * sometimes present. Call this method to get the list of [MusicMetadata]s that omits the shuffle
+	 * play button.
+	 */
+	private fun removeShufflePlayButtonMetadata(items: List<MusicMetadata>): List<MusicMetadata> {
+		return if (items.isNotEmpty() && items[0].artist == "" && items[0].title == L.MUSIC_SHUFFLE_PLAY_BUTTON_TITLE) {
```

Comment:
I like this solution, comparing the menu entry to a translated string to know when to omit it. However, the string comes from Spotify and follows the phone's translations, while the `L` object might not (defaults to the car's language, and can be manually set to any supported language).
Instead of adding to the `L` object, this should use `context.getString()` to load the phone language's translation of the string (perhaps at object initialization time). The string id should then use snake_case to match the other phone strings.

Reply:
Alternativelly, is there some clue from the ShufflePlay entry's uri that can be used, instead of the name?

Reply:
In my test just now, the Shuffle Play's uri is the same as the `queueUri`, which might hold true in general and be a good way to drop that entry.

Reply:
Ah good idea, I didn't realize the queueUri was the same as the shuffle play's uri! I removed the title matching and string and now only check if the mediaId == queueUri which is much better than before so now it is language agnostic